lib/features/popup-menu/PopupMenu.js

```diff
@@ -501,8 +504,16 @@ PopupMenu.prototype._createEntries = function(entries, className) {
   domClasses(entriesContainer).add(className);
 
   forEach(entries, function(entry, id) {
-    var entryContainer = self._createEntry(entry, id);
-    entriesContainer.appendChild(entryContainer);
+    var entryContainer = self._createEntry(entry, id),
+        grouping = entry.group || 'default',
+        groupContainer = domQuery('[data-group=' + grouping + ']', entriesContainer);
+
+    if (!groupContainer) {
+      groupContainer = domify('<div class="group" data-group="' + grouping + '"></div>');
```

Comment:
We should escape `grouping` to not allow XSS.

Reply:
Valid point. We will need to fix it in [the context pad](https://github.com/bpmn-io/diagram-js/blob/develop/lib/features/context-pad/ContextPad.js#L286) as well.

Reply:
This is now fixed, but there are also a couple of other places where we allow DOM injection. I will fix them.

Reply:
Woops woops :).